### PR TITLE
[MRG] Statistical mapping

### PR DESCRIPTION
**Pull request by bthirion**
Added elementary utilities for thresholding. Addresses part of #28

### CODE REVIEW

File: examples/plot_thresholding.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# perform a one-sample test on these values
+from scipy.stats import ttest_1samp
+_, p_values = ttest_1samp(fmri_masked, 0)
+
+# z-transform of p-values
+from nistats.utils import z_score
+z_map = nifti_masker.inverse_transform(z_score(p_values))
+
+# Threshold the resulting map:
+# false positive rate < 10^{-3}, cluster size > 10 voxels
```
Comment (MartinPerez) [inline examples/plot_thresholding.py]: 10^{-3} why not write 0.001

File: nistats/tests/test_thresholding.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    mask_img = nib.Nifti1Image(np.ones(shape), np.eye(4))
+
+    # test 1
+    th_map = map_threshold(
+        stat_img, mask_img, threshold, height_control='fpr',
+        cluster_threshold=0)
+    vals = th_map.get_data()
+    assert_equal(np.sum(vals > 0), 8)
+
+    # test 2:excessive size threshold
```
Comment (MartinPerez) [inline nistats/tests/test_thresholding.py]: The comment on test 2 and test 3 should be swapped?

File: nistats/tests/test_thresholding.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    # test 4: fdr threshold
+    for control in ['fdr', 'bonferoni']:
+        th_map = map_threshold(
+            stat_img, mask_img, .05, height_control='fdr',
+            cluster_threshold=5)
+        vals = th_map.get_data()
+        assert_equal(np.sum(vals > 0), 8)
+
+    # test 5: direct threshold
```
Comment (MartinPerez) [inline nistats/tests/test_thresholding.py]: missing test 6 for bonferroni threshold?

File: nistats/tests/test_thresholding.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    th_map = map_threshold(
+        stat_img, mask_img, 5., height_control=None,
+        cluster_threshold=5)
+    vals = th_map.get_data()
+    assert_equal(np.sum(vals > 0), 8)
```
Comment (MartinPerez) [inline nistats/tests/test_thresholding.py]: This line seems to be the origin of the travis fail? We get 0 voxels over threshold with that direct threshold value...

**Comment by MartinPerez** [comment]
All seems good to me, in case its useful I commented the line causing the travis failure.

**Comment by bthirion** [comment]
This one should be good to go. Any more comment ?

File: nistats/tests/test_thresholding.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
+""" Test the thresholding utilities
+"""
+import numpy as np
+from scipy.stats import norm
+from nose.tools import assert_true
+from numpy.testing import assert_almost_equal, assert_equal
+import nibabel as nib
+from ..thresholding import (fdr_threshold, map_threshold)
```
Comment (agramfort) [inline nistats/tests/test_thresholding.py]: don't use relative imports in tests

**Comment by agramfort** [comment]
besides LGTM

**Comment by bthirion** [comment]
The travis failure is unrelated: it comes from the nilearn change and should be handled in a separate PR. 
Good to go ?

File: nistats/thresholding.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        z_th = norm.isf(threshold / n_voxels)
+    else:  # Brute-force thresholding
+        z_th = threshold
+    stats *= (stats > z_th)
+
+    stat_map = masker.inverse_transform(stats).get_data()
+
+    # Extract connected components above threshold
+    label_map, n_labels = label(stat_map > z_th)
+    labels = label_map[(masker.mask_img.get_data() > 0)]
```
Comment (MartinPerez) [inline nistats/thresholding.py]: This line fails for a mask_img that is a path to a nifti file.

Comment (bthirion) [inline nistats/thresholding.py]: Fixed, thx for the suggestion. I haven't found yet any more elegant way to do it.

**Comment by MartinPerez** [comment]
I think its fine to merge

File: examples/plot_thresholding.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+z_map = nifti_masker.inverse_transform(z_score(p_values))
+
+# Threshold the resulting map:
+# false positive rate < .001, cluster size > 10 voxels
+from nistats.thresholding import map_threshold
+thresholded_map1 = map_threshold(
+    z_map, nifti_masker.mask_img_, threshold=.001, height_control='fpr',
+    cluster_threshold=10)
+
+# Now use FDR <.05, no lcuster-level threshold
```
Comment (lesteve) [inline examples/plot_thresholding.py]: typo: lcuster -> cluster

Comment (lesteve) [inline examples/plot_thresholding.py]: BTW you are using threshold=.01 and the comment says FDR < .05, which one is right?

Comment (bthirion) [inline examples/plot_thresholding.py]: Thx. This should be fixed now.

**Comment by bthirion** [comment]
If there is no further comment on that one, I'll merge it tomorrow.

**Pull request merged**